### PR TITLE
Organize app settings by category for easier reading in docs

### PR DESCRIPTION
**Pull request by esthersweon**
**User-Facing Changes**
Organized app settings by category and order in UI, for easier reference when included in the docs for the extension panel API

### CODE REVIEW

File: packages/studio-base/src/AppSetting.ts [hunk 1]
```diff
@@ -3,22 +3,33 @@
...
-  CRASH_REPORTING_ENABLED = "telemetry.crashReportingEnabled",
+  // General
+  COLOR_SCHEME = "colorScheme",
+  TIMEZONE = "timezone",
+  TIME_FORMAT = "time.format",
   MESSAGE_RATE = "messageRate",
-  LAUNCH_PREFERENCE = "launchPreference",
+  UPDATES_ENABLED = "updates.enabled",
+
+  //ROS
```
Comment (jhurliman) [inline packages/studio-base/src/AppSetting.ts:13]:
```suggestion
  // ROS
```

**Review by jtbandes — APPROVED** [review]

File: packages/studio-base/src/AppSetting.ts [hunk 2]
```diff
@@ -3,22 +3,33 @@
...
+  TIMEZONE = "timezone",
+  TIME_FORMAT = "time.format",
   MESSAGE_RATE = "messageRate",
-  LAUNCH_PREFERENCE = "launchPreference",
+  UPDATES_ENABLED = "updates.enabled",
+
+  // ROS
   ROS_PACKAGE_PATH = "ros.ros_package_path",
+
+  // Privacy
```
Comment (defunctzombie) [inline packages/studio-base/src/AppSetting.ts:16]:
```suggestion
  // Telemetry
```

File: packages/studio-base/src/AppSetting.ts [hunk 3]
```diff
@@ -3,22 +3,33 @@
...
   ENABLE_MCAP_DATA_SOURCE = "sources.mcap",
+
+  // Miscellaneous
+  HIDE_SIGN_IN_PROMPT = "hideSignInPrompt",
+  LAUNCH_PREFERENCE = "launchPreference",
   OPEN_DIALOG = "ui.open-dialog",
-  UPDATES_ENABLED = "updates.enabled",
   SHOW_OPEN_DIALOG_ON_STARTUP = "ui.open-dialog-startup",
+
+  // Dev only
```
Comment (defunctzombie) [inline packages/studio-base/src/AppSetting.ts:32]:
```suggestion
  // Debug
```

File: packages/studio-base/src/AppSetting.ts [hunk 4]
```diff
@@ -3,22 +3,33 @@
...
+
+  // Privacy
   TELEMETRY_ENABLED = "telemetry.telemetryEnabled",
-  TIME_FORMAT = "time.format",
-  TIMEZONE = "timezone",
+  CRASH_REPORTING_ENABLED = "telemetry.crashReportingEnabled",
+
+  // Experimental features
   UNLIMITED_MEMORY_CACHE = "experimental.unlimited-memory-cache",
   SHOW_DEBUG_PANELS = "showDebugPanels",
```
Comment (defunctzombie) [inline packages/studio-base/src/AppSetting.ts:22]: move SHOW_DEBUG_PANELS to the `Debug` section?

File: packages/studio-base/src/AppSetting.ts (hunk 2 above)
Comment (esthersweon) [reply packages/studio-base/src/AppSetting.ts:16]: Should this heading also be changed to Telemetry?

Screen Shot 2022-02-09 at 2 09 06 PM

File: packages/studio-base/src/AppSetting.ts (hunk 3 above)
Comment (esthersweon) [reply packages/studio-base/src/AppSetting.ts:32]: I want to clarify that these only show up in the dev environment. You can technically do other debugging things (like showing debug panels) in prod.

File: packages/studio-base/src/AppSetting.ts (hunk 4 above)
Comment (esthersweon) [reply packages/studio-base/src/AppSetting.ts:22]: Same comment as above.

Comment (defunctzombie) [reply packages/studio-base/src/AppSetting.ts:22]: 👍

File: packages/studio-base/src/AppSetting.ts (hunk 2 above)
Comment (defunctzombie) [reply packages/studio-base/src/AppSetting.ts:16]: I didn't realize that - ok. Consistent with what is in app sounds good.

**Pull request merged**